src/FakeItEasy/Configuration/AnyCallCallRule.cs

```diff
@@ -37,11 +37,22 @@ public override void UsePredicateToValidateArguments(Func<ArgumentCollection, bo
 
         protected override bool OnIsApplicableTo(IFakeObjectCall fakeObjectCall)
         {
-            return
-                this.argumentsPredicate(fakeObjectCall.Arguments) &&
-                    (this.ApplicableToMembersWithReturnType == null
-                    || (this.ApplicableToMembersWithReturnType == fakeObjectCall.Method.ReturnType)
-                    || (this.ApplicableToAllNonVoidReturnTypes && fakeObjectCall.Method.ReturnType != typeof(void)));
+            if (!this.argumentsPredicate(fakeObjectCall.Arguments))
+            {
+                return false;
+            }
+
+            if (this.ApplicableToAllNonVoidReturnTypes)
+            {
+                return fakeObjectCall.Method.ReturnType != typeof(void);
+            }
+
+            if (this.ApplicableToMembersWithReturnType != null)
+            {
+                return this.ApplicableToMembersWithReturnType == fakeObjectCall.Method.ReturnType;
+            }

```

Comment:
I'm wondering... can we have a situation where both `ApplicableToAllNonVoidReturnTypes` and `ApplicableToMembersWithReturnType != null` are true? I don't think so, but assuming it's possible, shouldn't we check `ApplicableToMembersWithReturnType` first, as it's more specific?